mne/beamformer/_dics.py

```diff
@@ -1,32 +1,556 @@
+        more).
+
+    Returns
+    -------
+    filters : dict
+        Dictionary containing filter weights from DICS beamformer.
+        Contains the following keys:
+
+            'weights' : ndarray, shape (n_frequencies, n_weights)
+                For each frequency, the filter weights of the beamformer.
+            'csd' : instance of CrossSpectralDensity
+                The data cross-spectral density matrices used to compute the
+                beamformer.
+            'ch_names' : list of str
+                Channels used to compute the beamformer.
+            'proj' : ndarray, shape (n_channels, n_channels)
+                Projections used to compute the beamformer.
+            'vertices' : list of ndarray
+                Vertices for which the filter weights were computed.
+            'mode' : 'scalar' | 'vector'
+                Whether the filters represent a scalar or vector beamformer.
+            'weight_norm' : None | 'unit-noise-gain'
+                The normalization of the weights.
+            'normalize_fwd' : bool
+                Whether the forward solution was normalized
+            'n_orient' : int
+                Number of source orientations defined in the forward model.
+            'subject' : str
+                The subject ID.
+
+    See Also
+    --------
+    apply_dics_csd, tf_dics
+
+    Notes
+    -----
+    The original reference is [1]_.
+
+    For more information about ``real_filter``, see the
+    `supplemental information <http://www.cell.com/cms/attachment/616681/4982593/mmc1.pdf>`_
+    from [3]_.
+
+    References
+    ----------
+    .. [1] Gross et al. Dynamic imaging of coherent sources: Studying neural
+           interactions in the human brain. PNAS (2001) vol. 98 (2) pp. 694-699
+    .. [2] Sekihara & Nagarajan. Adaptive spatial filters for electromagnetic
+           brain imaging (2008) Springer Science & Business Media
+    .. [3] Hipp JF, Engel AK, Siegel M (2011) Oscillatory Synchronization
+           in Large-Scale Cortical Networks Predicts Perception.
+           Neuron 69:387-396.
+    """  # noqa: E501
+    allowed_ori = [None, 'normal', 'max-power']
+    if pick_ori not in allowed_ori:
+        raise ValueError('"pick_ori" should be one of %s.' % allowed_ori)
+
+    # Leadfield rank and optional rank reduction
+    if reduce_rank and not (pick_ori == 'max-power' and mode == 'scalar'):
+        raise NotImplementedError(
+            'The computation of spatial filters with rank reduction using '
+            'reduce_rank parameter is only implemented with '
+            'pick_ori=="max-power" and mode="scalar".'
+        )
+
+    frequencies = [np.mean(freq_bin) for freq_bin in csd.frequencies]
+    n_freqs = len(frequencies)
+    n_orient = forward['sol']['ncol'] // forward['nsource']
+
+    # Determine how to normalize the leadfield
+    if normalize_fwd:
+        if mode == 'scalar':
+            leadfield_norm = 'point'
```

Comment:
How the leadfield is normalized is something the users might want to choose themselves, compare e.g. [Fieldtrip](https://github.com/fieldtrip/fieldtrip/blob/master/forward/ft_compute_leadfield.m#L569).